lib/routes/route_stats.ex

```diff
@@ -88,6 +242,50 @@ defmodule Routes.RouteStats do
     end)
   end
 
+  @spec instantaneous_headway_secs_of_vehicles([Vehicle.t()]) :: [integer()]
+  defp instantaneous_headway_secs_of_vehicles(vehicles) do
+    Enum.reduce(vehicles, [], fn vehicle, acc ->
+      case Vehicle.instantaneous_headway_secs(vehicle) do
```

Comment:
Ended up refactoring this whole bit, I think it's a little more readable now. I updated some tests because the previous method was reversing the vehicle stats from the order they came in as. (I assumed this was ok to change, let me know if not and I can pop an Enum.reverse on there.)